### PR TITLE
[ADD] add shopinvader_assortment addon

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
depends on https://github.com/OCA/product-attribute/pull/376

### CODE REVIEW

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <record id="shopinvader_assortment1" model="ir.filters">
+        <field name="name">Demo Shopinvader Backend</field>
```
Comment (lmignon) [inline shopinvader_assortment/demo/shopinvader_assortment_demo.xml:7]: could be "Shopinvader Assortment"'

File: shopinvader_assortment/models/shopinvader_backend.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        comodel_name="ir.filters",
+        help="Bind only products matching with the assortment domain",
+        domain=[('is_assortment', '=', True)],
+        context={
+            'product_assortment': True,
+            'default_is_assortment': 1},
+    )
+
+    @api.multi
+    def _products_autobinding_from_assortment(self):
```
Comment (lmignon) [inline shopinvader_assortment/models/shopinvader_backend.py]: why not `autobind_product_from_assortment'

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        <field name="model_id">product.product</field>
+        <field name="domain">[('sale_ok', '=', True)]</field>
+        <field name="user_id" eval="False"/>
+        <field name="is_assortment" eval="True"/>
```
Comment (lmignon) [inline shopinvader_assortment/demo/shopinvader_assortment_demo.xml:11]: Is this field `is_assortment` defined on `ir.filter`?

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml (hunk 1 above)
Comment (lmignon) [reply shopinvader_assortment/demo/shopinvader_assortment_demo.xml:7]: oups, may be defined in `product_assortment`

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml (hunk 3 above)
Comment (Cedric-Pigeon) [reply shopinvader_assortment/demo/shopinvader_assortment_demo.xml:11]: yes in product_assortment addon

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml (hunk 1 above)
Comment (Cedric-Pigeon) [reply shopinvader_assortment/demo/shopinvader_assortment_demo.xml:7]: yes but not really useful to do that, isn't it ?

File: shopinvader_assortment/models/shopinvader_backend.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                'backend_id': self.id,
+                'product_ids': [(6, 0, products_to_bind.ids)]
+            })
+            binding_wizard.bind_products()
+
+        if variants_to_unbind:
+            unbinding_wizard = unbinding_wizard_obj.create({
+                'shopinvader_variant_ids': [(6, 0, variants_to_unbind.ids)]
+            })
+            unbinding_wizard.unbind_products()
```
Comment (lmignon) [inline shopinvader_assortment/models/shopinvader_backend.py:51]: :heart: This code is as readable as reading a story to my children :smirk:

File: shopinvader_assortment/demo/shopinvader_assortment_demo.xml (hunk 3 above)
Comment (lmignon) [reply shopinvader_assortment/demo/shopinvader_assortment_demo.xml:11]: indeed I answered to the my comment in an other comment .... https://github.com/akretion/odoo-shopinvader/pull/216/files#r219508550

File: shopinvader_assortment/models/shopinvader_backend.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    _inherit = 'shopinvader.backend'
+
+    product_assortment_id = fields.Many2one(
+        comodel_name="ir.filters",
+        help="Bind only products matching with the assortment domain",
+        domain=[('is_assortment', '=', True)],
+        context={
+            'product_assortment': True,
+            'default_is_assortment': 1},
```
Comment (lmignon) [inline shopinvader_assortment/models/shopinvader_backend.py]: `default_is_assortment`is computed by the default method in `product_assortment` IMO this default should be removed

**Review by lmignon — APPROVED** [review]
Looks great and so useful/powerful

**Pull request merged**